money-core/src/main/scala/com/comcast/money/core/internal/SpanLocal.scala

```diff
@@ -37,7 +37,7 @@ trait SpanContext {
 object SpanLocal extends SpanContext {
 
   // A stack of span ids for the current thread
-  private[this] val threadLocalCtx = new ThreadLocal[Stack[Span]]
+  private[this] val threadLocalCtx = new ThreadLocal[List[Span]]
```

Comment:
I think we want this mutable, using ArrayBuffer

Reply:
The `ThreadLocal` itself is mutable, I'm replacing its value with the updated `List[Span]` on each push/pop.  The net effect is the same.  But if you'd prefer a mutable collection I can change the implementation.

Reply:
https://docs.scala-lang.org/overviews/collections/performance-characteristics.html

Reply:
ListBuffer will have the same performance characteristics as list though

https://www.scala-lang.org/api/current/scala/collection/mutable/ListBuffer.html

Reply:
after talking with @HaloFour  I don't think we gain anything from using a mutable collection. I mostly feel like analyzing performance without metrics won't glean us a whole lot either.

Reply:
@ptravers I agree, using a linked list is good from an allocation perspective